### ISSUE TITLE
Make `positional-arguments` default to true next edition

### ISSUE THREAD

**Issue by casey** [casey/just#1367]
To recap, `set positional-arguments` allows accessing arguments with`$N`:

```
$ cat justfile
set positional-arguments

foo bar:
  echo $1
$ just foo x
echo $1
x
```

This can be very convenient, since it lets you avoid word splitting and possibly extra layers of quoting. I wanted to make this default to true when it was initially implemented, since it's very useful. However, it used to break powershell, and since powershell is widely used, this wouldn't be a suitable default.

However, it appears that powershell now allows this:

```
$ pwsh --version
PowerShell 7.2.6
$ cat justfile
set shell := ['pwsh', '-Command']

set positional-arguments

foo bar:
  echo $1
$ just foo x
echo $1
x
```

So, `positional-arguments` is again a contender to default to true when the next edition is released.

**Comment by eRaMvn** [comment]
My apologies, but from your code 

```
$ cat justfile
set shell := ['pwsh', '-Command']

set positional-arguments

foo bar:
  echo $1
$ just foo x
echo $1
foo
x
```

Should this be the expected behavior?

```
$ just foo x
echo $1
x
```

Printing $1 should only prints `x` right

**Comment by casey** [comment]
@eRaMvn Good catch! Yup, that's correct, I fixed it.

**Comment by eRaMvn** [comment]
so when I played around with powershell, the output is exactly what you printed above
```
$ just foo x
echo $1
foo
x
```
so maybe there is a bug

**Comment by casey** [comment]
Weird! It doesn't do this with sh:

```just
set positional-arguments

foo bar:
  echo $1
```

```
$ just foo x
echo $1
x
```

What about if you do:

```
foo bar:
  echo "$1"
```

I'm curious if for some reason `$1` is including the recipe name, which should be in `$0` and not `$1`.

**Comment by eRaMvn** [comment]
Yeah, I think there is a bug for pwsh. I don't know rust well enough to make a PR 
<img width="355" alt="image" src="https://user-images.githubusercontent.com/28853937/208484043-00e3455e-59c0-4829-a059-7fa5efed77d6.png">

When I do `"$1"` the behavior is the same
<img width="340" alt="image" src="https://user-images.githubusercontent.com/28853937/208484266-75a7a184-9444-49f6-8d71-7a108ee89a8e.png">

Same for $0
<img width="353" alt="image" src="https://user-images.githubusercontent.com/28853937/208484381-99b6a98c-305b-41d8-bf66-1d4b63cb0fdf.png">

**Comment by cspotcode** [comment]
This is not a pwsh bug per-se, it's a misunderstanding of how positionals work.  I wrote about it in detail in #1592, I hope it helps.

The usefulness of `positional-arguments` relies on assumptions about bash and friends that are not true for powershell.  It assumes common posix `$@`, `$1`, syntax for positionals.  It assumes the shell will accept a command as a single string (single entry of its argv array) and keep it separate from any trailing positionals, capturing them into an `$@` array.  Powershell does not do this at all.  When we think that it's accepting positionals, it's not.  Rather, it's combining all positionals into a single command, interpreting all of it as powershell syntax to be executed.
